### DISCUSSION TITLE
IKEv2 support for xauth-pam with mschapv2

### DISCUSSION THREAD

**Discussion by gr4yg00s3** [strongswan/strongswan#1505] [General]
From the documentation I understand that the ability to continue using xauth-pam (with the IKEv1 pam configuration) with IKEv2, one needs to use eap-gtc as the authentication mechanism, as eap-gtc provides access to the cleartext password. 

I was wondering, that given mschapv2 is similar, and like with gtc the channel of communication would be protected by the server authentication via cert, is there a reason we cannot use a pam module with mschapv2 as well?

The reason I ask this is because almost all major OS's don't support eap-gtc

**Comment by tobiasbrunner** [comment]
> that given mschapv2 is similar

It's not similar at all. The client hashes the password locally with a challenge provided by the server. To verify that the server needs the plaintext password (or the initial NTLM hash). This won't work with PAM.

**Reply by gr4yg00s3** [reply]
oh okay yes, that makes sense. Is there any other way to use a pam module with IKEv2 other than gtc?

Edit: I was under the impression mschapv2 provides the NT hash

**Reply by tobiasbrunner** [reply]
I'm not aware of any.
